ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -61,3 +83,39 @@ func readECSParams(filename string) (*ECSParams, error) {
 
 	return ecsParams, nil
 }
+
+// ConvertToEcsNetworkConfiguration extracts out the NetworkConfiguration from
+// the ECSParams into a format that is compatible with ECSClient calls.
+func ConvertToEcsNetworkConfiguration(ecsParams *ECSParams) (*ecs.NetworkConfiguration, error) {
+	networkConfig := ecsParams.RunParams.NetworkConfiguration
+	awsvpcConfig := networkConfig.AwsVpcConfiguration
+	subnets := awsvpcConfig.Subnets
+	securityGroups := awsvpcConfig.SecurityGroups
+	networkMode := ecsParams.TaskDefinition.NetworkMode
+
+	if len(subnets) < 1 && networkMode == "awsvpc" {
+		return nil, errors.New("At least one subnet is required in the network configuration.")
```

Comment:
nit: this doesn't conform to https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
fixed